src/Template/Bake/tests/test_case.ctp

```diff
@@ -89,6 +89,7 @@ class <%= $className %>Test extends TestCase
     public function tearDown()
     {
         unset($this-><%= $subject %>);
+        TableRegistry::clear();
```

Comment:
@markstory I think it's fair to assume that controllers are very likely to use tables in most of the cases, and from personal experience I know that it's very easy to trip over this in integration tests.

Take this example, on the second test things will break, because the associations do not exist anymore. So my vote would be, table, controller, and maybe even entity tests.

```php
<?php
namespace App\Controller;

use App\Controller\AppController;

class ArticlesController extends AppController
{
    public function first()
    {
        $this->Articles->associations()->removeAll();
    }

    public function second()
    {
        $this->Articles->find()->contain('Tags')->all();
    }
}
```

```php
<?php
namespace App\Test\TestCase\Controller;

use Cake\TestSuite\IntegrationTestCase;

class ArticlesControllerTest extends IntegrationTestCase
{
    public $fixtures = [
        'app.articles'
    ];

    public function testFirst()
    {
        $this->get('/articles/first');
    }

    public function testSecond()
    {
        $this->get('/articles/second');
    }
}
```